Content.Shared/Weapons/Ranged/Systems/SharedGunSystem.Container.cs

```diff
@@ -32,7 +32,13 @@ private void OnContainerTakeAmmo(EntityUid uid, ContainerAmmoProviderComponent c
             if (_netMan.IsServer)
                 container.Remove(ent);
 
-            args.Ammo.Add((ent, EnsureComp<AmmoComponent>(ent)));
+            // if it's meant to throw items not shoot, ignore the ammo's cartridge component to prevent e.g. pneumatic cannon shooting bullets instead of throwing
+            if (!TryComp<AmmoComponent>(ent, out var ammo) || component.ThrowItems)
+            {
+                ammo = new AmmoComponent();
+            }
+
+            args.Ammo.Add((ent, ammo));
```

Comment:
Dont use component ctors like this, add it like it was previously

Reply:
its explicitly not adding the component to the ammo entity so that launching it doesn't shoot it thats the whole point of the pr

Reply:
The reason this is a problem to begin with is because in `Shoot()` the cartridge goes to the CartridgeComponent case instead of the ammo case  
![image](https://user-images.githubusercontent.com/60196617/227396146-4f6cef2f-7f47-4f33-852a-3209c4a2b524.png)

There are many ways to fix this, like making a bool somewhere in the call stack but making a dummy component with a ctor is not one of them

Reply:
So should I move the ammocomponent case into a function then call that if a bool arg is true? Or is most of it not needed?

Reply:
No, you should add a bool to CartridgeAmmoComponent (or a new optional arg in Shoot()) and add a TryThrow like in the ammo case to it i think